lib/krane/kubernetes_resource.rb

```diff
@@ -87,7 +87,7 @@ def validate_definition_essentials(definition)
           raise InvalidTemplateError.new("Template is missing required field 'kind'", content: debug_content)
         end
 
-        if definition.dig('metadata', 'name').blank?
+        if definition.dig('metadata', 'name').blank? && definition.dig('metadata', 'generateName').blank?
           raise InvalidTemplateError.new("Template is missing required field 'metadata.name'", content: debug_content)
```

Comment:
This error message is now out of date, it should be something like:
`Template has neither metadata.name nor metadata.generateName, but must specify exactly 1`

Also, what happens when _both_ name and generateName are specified: that's also an error case, no?

Reply:
Yes that's an error 👍 Good call.